yolov7/detect_ptg.py

```diff
@@ -53,18 +56,31 @@ def data_loader(recipes, split):
     videos = training_split[split]
     return videos
 
-def read_image(image_fn, imgsz, stride, device, half):
-    """Read the image file and preprocess it
 
-    :return:
-        - img0: The original image
-        - img: The preprocessed image
+def preprocess_bgr_img(
+    img_bgr: npt.ArrayLike,
+    imgsz: Union[int, Tuple[int, int]],
+    stride: int,
+    device: torch.device,
+    half: bool,
+) -> torch.Tensor:
+    """
+    Prepare a BGR image matrix for inference processing by the model.
+
+    :param img_bgr: Input BGR image matrix in [H, W, C] order.
+    :param imgsz: Integer (square size) or [height, width] tuple specifying the
```

Comment:
Is this definition coming from another function? The input parameter to the training script (https://github.com/cameron-a-johnson/yolov7/blob/e09d89bed32f2c82538885bb52c79aa0ecca4bee/yolov7/train.py#L540) reads to me like it would be a square shape and there doesn't seem to be a way to pass in a tuple to inference, fwiw I gave the training --image-size 1280 1280

Reply:
`imgsz` is passed into the `letterbox` function here, so I dove into that to understand what it wants there, and it can take in either an integer *or* a tuple:
```python
python-tpl/yolov7/yolov7/utils/datasets.py